src/database/database-common/src/db_transaction_manager.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::any::Any;
+use std::sync::Arc;
+
+use dill::{Catalog, CatalogBuilder};
+use internal_error::InternalError;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatabaseTransactionManager: Send + Sync {
+    async fn make_transaction(&self) -> Result<Transaction, InternalError>;
+
+    async fn commit_transaction(&self, transaction: Transaction) -> Result<(), InternalError>;
+
+    async fn rollback_transaction(&self, transaction: Transaction) -> Result<(), InternalError>;
```

Comment:
The error handling needs are a bit unclear right now. What I wanted to avoid is including database-aware error type into abstract repository, but maybe it's not a big deal, especially if we replace in-memory db with SQLite.

Reply:
Does not look like we have a luxury of detecting "optimistic concurrency". In the best case we will be getting a string, specific to database.

https://github.com/launchbadge/sqlx/blob/main/sqlx-core/src/error.rs